src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -1201,6 +1201,247 @@ public static void Main()
+        IL_000c: ret
+    }
+
+    .method public newslot virtual instance bool Equals ( class Base '' ) cil managed
+    {
+        IL_0000: ldarg.1
+        IL_0001: brfalse.s IL_002d
+
+        IL_0003: ldarg.0
+        IL_0004: callvirt instance class [mscorlib]System.Type Base::get_EqualityContract()
+        IL_0009: ldarg.1
+        IL_000a: callvirt instance class [mscorlib]System.Type Base::get_EqualityContract()
+        IL_000f: call bool [mscorlib]System.Type::op_Equality(class [mscorlib]System.Type, class [mscorlib]System.Type)
+        IL_0014: brfalse.s IL_002d
+
+        IL_0016: call class [mscorlib]System.Collections.Generic.EqualityComparer`1<!0> class [mscorlib]System.Collections.Generic.EqualityComparer`1<int32>::get_Default()
+        IL_001b: ldarg.0
+        IL_001c: ldfld int32 Base::'<i>k__BackingField'
+        IL_0021: ldarg.1
+        IL_0022: ldfld int32 Base::'<i>k__BackingField'
+        IL_0027: callvirt instance bool class [mscorlib]System.Collections.Generic.EqualityComparer`1<int32>::Equals(!0, !0)
+        IL_002c: ret
+
+        IL_002d: ldc.i4.0
+        IL_002e: ret
+    }
+
+    .method family hidebysig specialname rtspecialname instance void .ctor ( class Base '' ) cil managed
+    {
+        IL_0000: ldarg.0
+        IL_0001: call instance void [mscorlib]System.Object::.ctor()
+        IL_0006: ldarg.0
+        IL_0007: ldarg.1
+        IL_0008: ldfld int32 Base::'<i>k__BackingField'
+        IL_000d: stfld int32 Base::'<i>k__BackingField'
+        IL_0012: ret
+    }
+
+    .method public hidebysig instance void Deconstruct ( [out] int32& i ) cil managed
+    {
+        IL_0000: ldarg.1
+        IL_0001: ldarg.0
+        IL_0002: call instance int32 Base::get_i()
+        IL_0007: stind.i4
+        IL_0008: ret
+    }
+
+    .method public hidebysig instance void Clone () cil managed
+    {
+        IL_0000: ldstr ""RAN""
+        IL_0005: call void [mscorlib]System.Console::Write(string)
+        IL_000a: ret
+    }
+
+    .property instance class [mscorlib]System.Type EqualityContract()
+    {
+        .get instance class [mscorlib]System.Type Base::get_EqualityContract()
+    }
+
+    .property instance int32 i()
+    {
+        .get instance int32 Base::get_i()
+        .set instance void modreq(System.Runtime.CompilerServices.IsExternalInit) Base::set_i(int32)
+    }
+}
+
+.class public auto ansi abstract sealed beforefieldinit System.Runtime.CompilerServices.IsExternalInit extends [mscorlib]System.Object
+{
+}
+";
+            var src = @"
+record R(int i) : Base(i);
+
+public class C
+{
+    public static void Main()
+    {
+        var r = new R(1);
+        r.Clone();
+    }
+}
+";
+
+            var comp = CreateCompilationWithIL(src, il, options: TestOptions.DebugExe, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "RAN");
+            // Note: we do load the Clone method from metadata
```

Comment:
That is an interesting case. What is the rationale for allowing this vs. marking this as an error in metadata import?

I don't necessarily think this is wrong but curious about the reasoning here.

Reply:
@jaredpar I'm open to revisiting. My rationale was:
(1) we're putting a restriction with the idea of lifting it in the future, it may not be worthwhile to be airtight to catch people who really go out of their way to do weird things.
(2) not knowing what the future design would be and whether it would require a new framework, I'm not sure we want to prevent the C# 9 compiler from using types produced by a future compiler (ie. that do have speakable `Clone` methods).